pkg/controller/envoy/ads_event_test.go

```diff
@@ -0,0 +1,792 @@
+		}
+		routeConfig := &config_route_v3.RouteConfiguration{
+			Name: "ut-routeconfig",
+			VirtualHosts: []*config_route_v3.VirtualHost{
+				{
+					Name: "ut-host",
+				},
+			},
+		}
+		anyRouteConfig, err := anypb.New(routeConfig)
+		assert.NoError(t, err)
+		rsp := &service_discovery_v3.DiscoveryResponse{
+			Resources: []*anypb.Any{
+				anyRouteConfig,
+			},
+		}
+		err = svc.handleRdsResponse(rsp)
+		assert.NoError(t, err)
+		wantHash := hash.Sum64String(anyRouteConfig.String())
+		actualHash := svc.DynamicLoader.RouteCache.GetRdsHash(routeConfig.GetName())
+		assert.Equal(t, wantHash, actualHash)
+		assert.Equal(t, []string{"ut-routeclient", "ut-routeconfig"}, svc.ack.ResourceNames)
+	})
+
+	t.Run("empty routeConfig", func(t *testing.T) {
+		adsLoader := NewAdsLoader()
+		svc := &ServiceEvent{
+			DynamicLoader: adsLoader,
+			ack: &service_discovery_v3.DiscoveryRequest{
+				ResourceNames: []string{
+					"ut-routeclient",
+				},
+			},
+		}
+		routeConfig := &config_route_v3.RouteConfiguration{}
+		anyRouteConfig, err := anypb.New(routeConfig)
+		assert.NoError(t, err)
+		rsp := &service_discovery_v3.DiscoveryResponse{
+			Resources: []*anypb.Any{
+				anyRouteConfig,
+			},
+		}
+		err = svc.handleRdsResponse(rsp)
+		assert.NoError(t, err)
+		wantHash := hash.Sum64String(anyRouteConfig.String())
+		actualHash := svc.DynamicLoader.RouteCache.GetRdsHash(routeConfig.GetName())
+		assert.Equal(t, wantHash, actualHash)
+		assert.Equal(t, []string{"ut-routeclient"}, svc.ack.ResourceNames)
+	})
+
+	t.Run("already have a Rds, RdsHash has change", func(t *testing.T) {
+		adsLoader := NewAdsLoader()
+		svc := &ServiceEvent{
+			DynamicLoader: adsLoader,
+			ack: &service_discovery_v3.DiscoveryRequest{
+				ResourceNames: []string{
+					"ut-routeclient",
+				},
+			},
+		}
+		routeConfig := &config_route_v3.RouteConfiguration{
+			Name: "ut-routeconfig",
+			VirtualHosts: []*config_route_v3.VirtualHost{
+				{
+					Name: "ut-host",
+				},
+			},
+		}
+		anyRouteConfig, err := anypb.New(routeConfig)
+		assert.NoError(t, err)
+		rsp := &service_discovery_v3.DiscoveryResponse{
+			Resources: []*anypb.Any{
+				anyRouteConfig,
+			},
+		}
+		err = svc.handleRdsResponse(rsp)
+		assert.NoError(t, err)
+
+		routeConfig.VirtualHosts = append(routeConfig.VirtualHosts, &config_route_v3.VirtualHost{Name: "new-ut-host"})
+		anyRouteConfig, err = anypb.New(routeConfig)
+		assert.NoError(t, err)
+		rsp = &service_discovery_v3.DiscoveryResponse{
+			Resources: []*anypb.Any{
+				anyRouteConfig,
+			},
+		}
+		svc.ack = newAckRequest(rsp)
+		err = svc.handleRdsResponse(rsp)
+		assert.NoError(t, err)
+		wantHash := hash.Sum64String(anyRouteConfig.String())
+		actualHash := svc.DynamicLoader.RouteCache.GetRdsHash(routeConfig.GetName())
+		assert.Equal(t, wantHash, actualHash)
+		assert.Equal(t, []string{"ut-routeconfig"}, svc.ack.ResourceNames)
+	})
+
+	t.Run("already have a Rda, RdaHash has change. And have mult routeconfig in rsp", func(t *testing.T) {
```

Comment:
`s/Rda/Rds/`,  `s/mult/multi/`, `s/rsp/resp/`, please check all the code to make sure all the typos are fixed.